internal/provider/resources/block.go

```diff
@@ -127,9 +129,64 @@ func (r *BlockResource) Create(ctx context.Context, req resource.CreateRequest,
 }
 
 // Read refreshes the Terraform state with the latest data.
-//
-//nolint:revive // TODO: remove this comment when method is implemented
 func (r *BlockResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state BlockResourceModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if state.ID.IsNull() {
+		resp.Diagnostics.AddError(
+			"ID is unset",
+			"This is a bug in the Terraform provider. Please report it to the maintainers.",
+		)
+	}
+
+	client, err := r.client.BlockDocuments(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating block client",
+			fmt.Sprintf("Could not create block client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	var blockID uuid.UUID
+	blockID, err = uuid.Parse(state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("id"),
+			"Error parsing Block ID",
+			fmt.Sprintf("Could not parse block ID to UUID, unexpected error: %s", err.Error()),
+		)
+
+		return
```

Comment:
it's worth mentioning here before we go too far down the rabbit hole - just like you have it here, anything that the user interfaces with here will be the `block`, whereas the API resources that constitute it are the `block_document` / `block_type` / `block_schema` etc.

Reply:
Good callout 👍🏼